code/mob/living/critter/ai/flock/flocktasks.dm

```diff
@@ -441,11 +443,10 @@ stare
 		F.hud?.update_hands() // for observers
 
 /datum/aiTask/sequence/goalbased/deposit/get_targets()
-	var/mob/living/critter/flock/drone/F = holder.owner
 	. = list()
-	for(var/obj/flock_structure/ghost/S in view(max_dist, F))
-		if(S.flock == F.flock && S.goal > S.currentmats)
-			. += S
+	for (var/obj/flock_structure/ghost/O as anything in by_type[/obj/flock_structure/ghost])
+		if (IN_RANGE(holder.owner, O, max_dist))
+			. += O
```

Comment:
I think this could break ghost structure depositing for if multiple flocks are in the game, assuming you don't want flock depositing to each other's structures?